pkgs/misc/arm-trusted-firmware/default.nix

```
@@ -17,22 +17,21 @@ let
             , platformCanUseHDCPBlob ? false  # set this to true if the platform is able to use hdcp.bin
             , extraMakeFlags ? []
             , extraMeta ? {}
-            , version ? "2.7"
             , ... } @ args:
 
            # delete hdcp.bin if either: the platform is thought to
            # not need it or unfreeIncludeHDCPBlob is false
            let deleteHDCPBlobBeforeBuild = !platformCanUseHDCPBlob || !unfreeIncludeHDCPBlob; in
 
-           stdenv.mkDerivation ({
+           stdenv.mkDerivation (finalAttrs: {
 
     pname = "arm-trusted-firmware${lib.optionalString (platform != null) "-${platform}"}";
-    inherit version;
+    version = "2.7";
 
     src = fetchFromGitHub {
       owner = "ARM-software";
       repo = "arm-trusted-firmware";
-      rev = "v${version}";
+      rev = "v${finalAttrs.version}";
```

```suggestion
      rev = "v${version}";
```
this is a fixed output derivation. Just changing rev does not invalidate the hash.

---

Wow, you're right.  Yet another footgun.